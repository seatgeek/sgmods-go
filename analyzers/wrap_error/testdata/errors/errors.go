package errors

import "fmt"

var ErrOpenFile = fmt.Errorf("error opening file")
var ErrReadFile = fmt.Errorf("error reading file")
var ErrFriday = fmt.Errorf("error - friday's are for fun, not files!")
