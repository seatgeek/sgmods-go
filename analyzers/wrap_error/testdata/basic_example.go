package testdata

import ( // want `adding "github.com/pkg/errors" import`
	"fmt"
	"log"
	"os"
	"time"
)

var ErrOpenFile = fmt.Errorf("error opening file")
var ErrReadFile = fmt.Errorf("error reading file")
var ErrFriday = fmt.Errorf("error - friday's are for fun, not files!")

func readFileBasicExample(name string) ([]byte, error) {
	file, err := os.Open(name)
	if err != nil {
		return nil, ErrOpenFile // want `unwrapped error found 'return nil, ErrOpenFile'`
	}

	var body []byte
	if _, err = file.Read(body); err != nil {
		return nil, ErrReadFile // want `unwrapped error found 'return nil, ErrReadFile'`
	}

	if time.Now().UTC().Weekday() == time.Friday {
		return nil, ErrFriday
	}

	return body, nil
}

func mainBasicExample() {
	body, err := readFileBasicExample(os.Args[1])
	if err != nil {
		log.Fatal(err)
	}

	fmt.Println(body)
}
