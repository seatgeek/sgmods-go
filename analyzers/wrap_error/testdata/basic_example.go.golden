package testdata

import (
	"fmt"
	"github.com/pkg/errors"
	"log"
	"os"
	"time"
)

var ErrOpenFile = fmt.Errorf("error opening file")
var ErrReadFile = fmt.Errorf("error reading file")
var ErrFriday = fmt.Errorf("error - friday's are for fun, not files!")

func readFileBasicExample(name string) ([]byte, error) {
	file, err := os.Open(name)
	if err != nil {
		return nil, errors.Wrap(ErrOpenFile, err.Error()) // want `unwrapped error found 'return nil, ErrOpenFile'`
	}

	var body []byte
	if _, err = file.Read(body); err != nil {
		return nil, errors.Wrap(ErrReadFile, err.Error()) // want `unwrapped error found 'return nil, ErrReadFile'`
	}

	if time.Now().UTC().Weekday() == time.Friday {
		return nil, ErrFriday
	}

	return body, nil
}

func mainBasicExample() {
	body, err := readFileBasicExample(os.Args[1])
	if err != nil {
		log.Fatal(err)
	}

	fmt.Println(body)
}
